### PR TITLE
Feature/grid layer

### PR DESCRIPTION
**Pull request by Brynjulf**
Base classes for layer and canvas layer, with an implementation for a grid layer. There are some things missing, like handling the loading, and setting and use the opacity. But this serves as a starting point
(#28)

### CODE REVIEW

File: .storybook/src/grid-layer.stories.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { scaleLinear } from 'd3-scale';
+import GridLayer from '../../src/layers/GridLayer';
+
+const width = 400;
+const height = 500;
+
+export default {
+  title: 'Canvas',
+};
+const xbounds = [0, 1000];
```
Comment (stalenestas) [inline .storybook/src/grid-layer.stories.ts]: group all constants

File: src/layers/CanvasLayer.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+abstract class CanvasLayer extends Layer {
+  ctx: CanvasRenderingContext2D;
+  elm: HTMLElement;
+
+  onMount(event: OnMountEvent) {
+    super.onMount(event);
+    this.elm = event.elm;
+
+    const canvas = select(event.elm).append('canvas')
```
Comment (stalenestas) [inline src/layers/CanvasLayer.ts]: Do we want d3 select in other that svg layer? Should we have some simpler alternative or just use plain js?

File: src/layers/GridLayer.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    this.render(event);
+  }
+
+  onRescale(event: OnRescaleEvent) {
+    super.onRescale(event);
+    this.render(event);
+  }
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
+
```
Comment (stalenestas) [inline src/layers/GridLayer.ts]: remove extra whitespace

File: src/layers/GridLayer.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    const {
+      ctx,
+      options,
+    } = this;
+
+    if (!ctx) return;
```
Comment (stalenestas) [inline src/layers/GridLayer.ts]: always brackets
if(!ctx) { return; }

File: src/layers/GridLayer.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+
+    const [rx1, rx2] = xscale.range();
+    const [ry1, ry2] = yscale.range();
+
+    const xticks = xscale.ticks();
+
+    let xminticks = [];
+    if (xticks.length >= 2) {
+      xminticks = xticks.map((v : any) => v + (xticks[1] - xticks[0]) / 2);
```
Comment (stalenestas) [inline src/layers/GridLayer.ts]: looks almost like the yminticks, refactor into method?

File: src/layers/GridLayer.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    // y major grid lines
+    yticks.forEach((ty : any) => {
+      const y = yscale(ty);
+      ctx.beginPath();
+      ctx.moveTo(rx1, y);
+      ctx.lineTo(rx2, y);
+      ctx.stroke();
+    });
+    ctx.restore();
```
Comment (stalenestas) [inline src/layers/GridLayer.ts]: refactor into methods and make the render much smaller

File: src/layers/GridLayer.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    // y minor grid lines
+    yminticks.forEach((ty : any) => {
+      const y = yscale(ty);
+      ctx.beginPath();
+      ctx.moveTo(rx1, y);
+      ctx.lineTo(rx2, y);
+      ctx.stroke();
+    });
+
+    ctx.lineWidth = options.majorWidth || 0.75;
```
Comment (stalenestas) [inline src/layers/GridLayer.ts]: refactor magic numbers to constants

File: src/layers/CanvasLayer.ts (hunk 2 above)
Comment (Brynjulf) [reply src/layers/CanvasLayer.ts]: used plain js

**Pull request merged**